14_2_Using_i2p.md

```diff
@@ -0,0 +1,50 @@
+# Chapter 14.2: Using i2p
+
+The Invisible Internet Project (I2P) is a fully encrypted private network layer. It uses a distributed [network database](https://geti2p.net/en/docs/how/network-database) and encrypted unidirectional tunnels between your and your peers.
+
+Basic differences between Tor and i2p: https://geti2p.net/en/comparison/tor
+
+Trade-offs: https://bitcoin.stackexchange.com/questions/107060/tor-and-i2p-tradeoffs-in-bitcoin-core
+
+It is not installed by [Bitcoin Standup](https://github.com/BlockchainCommons/Bitcoin-Standup-Scripts) right now as i2p support was recently added in Bitcoin core. However, you can try it manually by following the below steps:
+
+1. Install `i2pd`: https://i2pd.readthedocs.io/en/latest/user-guide/install/
+2. Run `i2pd`: [`sudo systemctl start i2pd.service`](https://i2pd.readthedocs.io/en/latest/user-guide/run/)
+3. Confirm if `i2pd` is running: `ss -nlt`
+
+   ```
+   State   Recv-Q   Send-Q     Local Address:Port      Peer Address:Port  Process
+
+   LISTEN  0        4096           127.0.0.1:7656           0.0.0.0:*
+   ```
+4. `bitcoin.conf`:
+
+   ```
+   testnet=1
+   i2psam=127.0.0.1:7656
+   debug=i2p
+   ```
+5. Run `bitcoind`
+
+6. Check `debug.log`:
+   ```
+   2021-06-15T20:36:16Z i2paccept thread start
+   2021-06-15T20:36:16Z I2P: Creating SAM session with 127.0.0.1:7656
+
+   2021-06-15T20:36:56Z I2P: SAM session created: session id=3e0f35228b, my address=bmwyyuzyqdc5dcx27s4baltbu6zw7rbqfl2nmclt45i7ng3ul4pa.b32.i2p:18333
+   2021-06-15T20:36:56Z AddLocal(bmwyyuzyqdc5dcx27s4baltbu6zw7rbqfl2nmclt45i7ng3ul4pa.b32.i2p:18333,4)
+   ```
+7. Confirm `i2p_private_key` was created:
+   ```
+   ~/.bitcoin/testnet3$ ls
+
+   anchors.dat  chainstate         i2p_private_key  settings.json
+   banlist.dat  debug.log          mempool.dat      wallets
+   blocks       fee_estimates.dat  peers.dat
+   ```
+8. `bitcoin-cli -netinfo` or `bitcoin-cli getnetworkinfo`returns i2p address:
+
+   ```
+   Local addresses
+   bmwyyuzyqdc5dcx27s4baltbu6zw7rbqfl2nmclt45i7ng3ul4pa.b32.i2p     port  18333    score      4
+   ```
```

Comment:
It might be this is still in process? But I feel like we need to see more about how to use i2p at this point, etc. (Or if *nothing* more is required, what it'll do and how it'll be useful.)

And then of course the standard conclusion that ends out each section.

Reply:
> It might be this is still in process? But I feel like we need to see more about how to use i2p at this point, etc. (Or if nothing more is required, what it'll do and how it'll be useful.)

Agree. Want to start with basic things and do follow up PRs to add more information as I learn more about i2p.

> And then of course the standard conclusion that ends out each section.

Okay